doc/implementation_comparison.md

```diff
@@ -0,0 +1,181 @@
+- Grpc.Core could provide an inprocess transport support but currently this functionality is not exposed in C# API.
+
+### Retries / Request Hedging
+
+Grpc.Core: implementation in C-core in progress, but no ETA yet
+
+grpc-dotnet: not implemented
+
+https://github.com/grpc/proposal/blob/master/A6-client-retries.md, 
+
+### Channelz
+
+Grpc.Core: not supported at the moment, but most of functionality is available in C-core through an API, a C# gRPC service needs to be implemented to expose the stats obtained from C-core as a gRPC service (relatively easy)
+
+grpc-dotnet: not supported
+
+https://github.com/grpc/proposal/blob/master/A14-channelz.md
+
+### Binary logging
+
+Grpc.Core: Implemented in C-core, but not exposed in the C# layer.
+grpc-dotnet: not implemented
+
+https://github.com/grpc/grpc/blob/master/doc/binary-logging.md
+
+### Compression
+
+Grpc.Core: supported (algorithms: `gzip`, `deflate`)
+
+grpc-dotnet: supported (algorithms: `gzip`), also provides public API to provide custom compression algorithm.
+
+Performance implications of using compression in both implementations haven't been measured. Compression functionality is offered mostly to comply with the spec.
+
+https://github.com/grpc/grpc/blob/master/doc/compression.md
+https://github.com/grpc/grpc/blob/master/doc/compression_cookbook.md
+
+### Fine-grained Transport Control: Connectivity API
+
+Grpc.Core: supported (exposed publicly on Channel), provided by C-Core
+
+grpc-dotnet: not supported (to provide support, changes to .NET HttpClient are needed, so adding support is non-trivial).
+
+https://github.com/grpc/grpc/blob/master/doc/connectivity-semantics-and-api.md
+
+### Fine-grained Transport Control: Connection Backoff
+
+Grpc.Core: supported, provided by C-core
+
+grpc-dotnet: not supported (HttpClient doesn't support)
+
+https://github.com/grpc/grpc/blob/master/doc/connection-backoff.md
+
+### Fine-grained Transport Control: Keepalive
+
+Grpc.Core: supported, provided by C-core
+
+grpc-dotnet: not supported (HttpClient and Kestrel don't provide support)
+
+https://github.com/grpc/grpc/blob/master/doc/keepalive.md
+
+### Fine-grained Transport Control: RPC Wait-for-ready
+
+Grpc.Core: supported, provided by C-core
+grpc-dotnet: not supported (implementing the wait_for_ready flag requires supporting channel connectivity first).
+
+https://github.com/grpc/grpc/blob/master/doc/wait-for-ready.md
+
+### Naming / Resolver API
+
+Grpc.Core: conforms with the spec thanks to the C-core dependency. It is currently not possible to provide custom C# resolvers via resolver API (the APIs are in C core and aren't exposed in the C# layer).
+
+grpc-dotnet: only resolves basic DNS names. No API to provide a custom resolver.
+
+https://github.com/grpc/grpc/blob/master/doc/naming.md
+
+### Port sharing (gRPC and HTTP traffic on the same port)
+
+grpc-dotnet allows port-sharing: serving both gRPC and non-gRPC traffic by the same server (Grpc.Core doesn't support)
+
+## Add-on Features
```

Comment:
Should add section on HttpClientFactory integration - https://docs.microsoft.com/en-us/aspnet/core/grpc/clientfactory?view=aspnetcore-3.0

Also discuss automatic cancellation and deadline propagation - https://docs.microsoft.com/en-us/aspnet/core/grpc/clientfactory?view=aspnetcore-3.0#deadline-and-cancellation-propagation

Reply:
Feel free to merge this PR and add these in a followup PR?